gaps/gap-25_golem_certificates/gap-25_golem_certificates.md

```diff
@@ -0,0 +1,181 @@
+In further documentation I assume we got number `1.3.6.1.4.1.60000`, until we will acquire real OID.
+
+Having our OID, we can define arbitrary semantic to numbers in our namespace:
+
+
+| Name                       | OID                      | Semantic                                                             |
+| -------------------------- | ------------------------ | -------------------------------------------------------------------- |
+| GolemFactory               | 1.3.6.1.4.1.60000 (TODO) | Root OID for Golem Factory                                           |
+| GolemCertificates          | ${GolemFactory}.1        | Root for Golem certificates parameters                               |
+| GolemCertificatesAllowance | ${GolemCertificates}.1   | List of permissions to sign certain Golem features using certificate |
+
+Example values of `GolemCertificatesAllowance` field (this list will be used in further examples, to illustrate how certification works):
+
+
+| Field             | Semantic                                                                        |
+| ----------------- | ------------------------------------------------------------------------------- |
+| all               | Certificate is allowed to sign any feature (even not existing future use cases) |
+| manifest-outbound | Allowed to sign manifests with outbound network enabled                         |
+| inbound           | Allowed to sign inbound allowance for Requestors.                               |
+| ...               | {List is non-exhaustive - new features will be created in future}               |
+
+### Permissions chains
+
+To transfer some set of permissions to external entity, Golem Factory has to sign it's certificate. This will create certificates chain, that will be later validated by Agent application. For signature to be valid, both certificate and it's parent has to contain specific allowance (or `all`).
+
+For example, if we have hierarchy:
+
+
+| Certificate           | allowance                 |
+| --------------------- | ------------------------- |
+| Golem Root            | all                       |
+| Golem Intermediate    | manifest-outbound,inbound |
+| Partner's certificate | manifest-outbound         |
+
+than Payload Manifest signature is valid and `Parter's certificate` is allowed to sign Manifests, but has no rights to sign inbound related features.
+
+But if structure looks like this:
+
+
+| Certificate           | allowance         |
+| --------------------- | ----------------- |
+| Golem Root            | all               |
+| Golem Intermediate    | inbound           |
+| Partner's certificate | manifest-outbound |
+
+Than `Parter's certificate` is not allowed to sign manifests and Agent application should reject such Proposals.
+Note that certificates chain is incorrect, because `Partner's certificate` shouldn't be signed by `Golem Intermediate` in the first place.
```

Comment:
```suggestion
Note that the certificate chain is incorrect, because `Partner's certificate` shouldn't be signed by `Golem Intermediate` in the first place.
```